### ISSUE TITLE
Prevent unnecessary parent-child traversal in require

### ISSUE THREAD

**Issue by simlu** [eslint/eslint#12109]
**Please describe what the rule should do:**

When a `require` references a relative file, but unnecessarily traverses up and down the file tree, this should be a warning.

**What category of rule is this? (place an "X" next to just one item)**

[X] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

Assuming the following code is found in `src/index.js`

```js
const misc = require('../src/util/misc');
```
The rule should warn and suggest to replace with `const misc = require('./util/misc');`

Indication of possible import error if e.g. the src folder gets compiled into a lib folder at some point.

**Why should this rule be included in ESLint (instead of a plugin)?**

It seems very general, so having it in eslint should be desirable.

**Are you willing to submit a pull request to implement this rule?**

Maybe at some point in the future. Time is very limited on my end atm.

**Comment by platinumazure** [comment]
Hi @simlu, thanks for the issue.

I'm not convinced this should be in ESLint core, simply because we have a lot of rules now and we would prefer to add new rules that are really high value for users. I'm not convinced the value here is worth the maintenance burden. Maybe other team members will feel differently.

I think this would be a great feature for [`eslint-plugin-import`](https://github.com/benmosher/eslint-plugin-import), personally.

**Comment by simlu** [comment]
As a reference, I've added this rule to the blackflux linting for now:
https://github.com/blackflux/eslint-plugin-rules/pull/264/files

**Comment by simlu** [comment]
Thank you for the feedback. I've now opened a new ticket here: https://github.com/benmosher/eslint-plugin-import/issues/1457

Closing...
